Automattic/woocommerce-payments
# Improve the logic which determines if a subscriptions plugin is being activated

#### Changes proposed in this Pull Request

Inside the function to check if we need to load subscriptions-core functionality we have a condition that checks if the subscriptions plugin is in the process of being activated. The purpose of this check is to avoid loading subscriptions core in the request that will later activate the Subscriptions plugin and cause a fatal errors.

This commit tightens the logic to avoid running these checks on all page loads as they were only intended to check when a user activates the plugin via the plugin screen.

#### Testing instructions

To confirm the intended use of this logic:

* Go to the Plugins screen.
* Optional. Deactivate WC Subscriptions and any subscriptions-core plugins if you have them active. 
* Activate the WC Subscriptions plugin.
* You shouldn't notice any fatal errors.
* If you deactivate this logic by commenting out the `return true;` and repeat the same steps, you will get a fatal. 

<img width="1944" alt="Screenshot 2023-06-22 at 1 49 28 pm" src="https://github.com/Automattic/woocommerce-payments/assets/8490476/6ea7645b-dcd5-474a-a66d-2de8bebe3dfc">

-------------------


**Post merge**
